src/udp/server/UDPServer.cpp

```diff
@@ -1,8 +1,57 @@
 // Copyright 2019 Lukas Joswiak, Justin Johnson, and Jack Khuu.
 
+#include "UDPServer.hpp"
+
 #include <iostream>
 
-int main() {
-  std::cout << "Hello, World!" << std::endl;
-  return 0;
+#include <boost/bind.hpp>
+
+UDPServer::UDPServer(boost::asio::io_context &io_context, int port)
+    : socket_(io_context, udp::endpoint(udp::v4(), port)) {
+  StartRead();
+}
+
+void UDPServer::StartRead() {
+  socket_.async_receive_from(boost::asio::buffer(recv_buffer_),
+      remote_endpoint_, boost::bind(&UDPServer::HandleRead, this,
+        boost::asio::placeholders::error,
+        boost::asio::placeholders::bytes_transferred));
+}
+
+void UDPServer::HandleRead(const boost::system::error_code &ec,
+    std::size_t bytes_transferred) {
+  if (!ec) {
+    auto message = std::string(&recv_buffer_[0],
+        &recv_buffer_[0] + bytes_transferred);
+    std::cout << "Received message: " << message << std::endl;
+
+    std::string response = "Hello, Client!";
+
+    socket_.async_send_to(boost::asio::buffer(response), remote_endpoint_,
+        boost::bind(&UDPServer::HandleSend, this,
+          boost::asio::placeholders::error,
```

Comment:
double indent?

Reply:
The second indent is due to calling another function, `bind`, inside the first indent.